src/pybind/mgr/volumes/fs/subvolume.py

```diff
@@ -252,56 +202,63 @@ def _get_ancestor_xattr(self, path, attr):
             else:
                 return self._get_ancestor_xattr(os.path.split(path)[0], attr)
 
-    def get_group_path(self, group_id):
-        path = self._group_path(group_id)
-        try:
-            self.fs.stat(path)
-        except cephfs.ObjectNotFound:
-            return None
-        return path
-
-    def get_subvolume_path(self, subvolume_path):
-        path = self._subvolume_path(subvolume_path)
-        try:
-            self.fs.stat(path)
-        except cephfs.ObjectNotFound:
-            return None
-        return path
-
-    def _snapshot_path(self, dir_path, snapshot_name):
-        return os.path.join(
-            dir_path, self.rados.conf_get('client_snapdir'), snapshot_name
-        )
-
-    def _snapshot_create(self, dir_path, snapshot_name, mode=0o755):
+    def _snapshot_create(self, snappath, mode=0o755):
         """
         Create a snapshot, or do nothing if it already exists.
         """
-        snapshot_path = self._snapshot_path(dir_path, snapshot_name)
         try:
-            self.fs.stat(snapshot_path)
+            self.fs.stat(snappath)
         except cephfs.ObjectNotFound:
-            self.fs.mkdir(snapshot_path, mode)
+            self.fs.mkdir(snappath, mode)
         else:
-            log.warn("Snapshot '{0}' already exists".format(snapshot_name))
-
+            log.warn("Snapshot '{0}' already exists".format(snappath))
 
-    def _snapshot_delete(self, dir_path, snapshot_name):
+    def _snapshot_delete(self, snappath):
         """
         Remove a snapshot, or do nothing if it doesn't exist.
         """
-        snapshot_path = self._snapshot_path(dir_path, snapshot_name)
-        self.fs.stat(snapshot_path)
-        self.fs.rmdir(snapshot_path)
+        self.fs.stat(snappath)
+        self.fs.rmdir(snappath)
 
-    def create_subvolume_snapshot(self, subvolume_path, snapshot_name, mode=0o755):
-        return self._snapshot_create(self._subvolume_path(subvolume_path), snapshot_name, mode)
+    def create_subvolume_snapshot(self, spec, snapname, mode=0o755):
```

Comment:
Why are subvolume snapshot create and rm methods under 'group operations' category?

Reply:
that's due to fall through from "groups" comment.